modules/product/src/Element/CommerceProductRenderedAttribute.php

```diff
@@ -0,0 +1,78 @@
+<?php
+
+namespace Drupal\commerce_product\Element;
+
+use Drupal\Core\Form\FormStateInterface;
+use Drupal\Core\Render\Element\Radios;
+use Drupal\Component\Utility\Html as HtmlUtility;
+
+/**
+ * Provides a form input element for rendering attributes as radio buttons.
+ *
+ * Example usage:
+ * @code
+ * $form['settings']['active'] = array(
+ *   '#type' => 'commerce_product_rendered_attribute',
+ *   '#title' => $this->t('Poll status'),
+ *   '#default_value' => 1,
+ *   '#options' => array(0 => $this->t('Closed'), 1 => $this->t('Active')),
+ * );
+ * @endcode
+ *
+ * @FormElement("commerce_product_rendered_attribute")
+ */
+class CommerceProductRenderedAttribute extends Radios {
+
+  /**
+   * Expands a radios element into individual radio elements.
+   */
+  public static function processRadios(&$element, FormStateInterface $form_state, &$complete_form) {
+    if (count($element['#options']) > 0) {
+
+      $element['#attached']['library'][] = 'commerce_product/rendered-attributes';
+
+      /** @var \Drupal\Core\Entity\EntityStorageInterface $storage */
+      $storage = \Drupal::service('entity_type.manager')->getStorage('commerce_product_attribute_value');
+      $attribute_values = $storage->loadMultiple(array_keys($element['#options']));
+
+      /** @var \Drupal\Core\Render\RendererInterface $renderer */
+      $renderer = \Drupal::service('renderer');
+
+      $weight = 0;
+      foreach ($element['#options'] as $key => $choice) {
+        $rendered_attribute = \Drupal::entityTypeManager()->getViewBuilder('commerce_product_attribute_value')->view($attribute_values[$key], 'add_to_cart');
```

Comment:
Let's move $view_builder =  \Drupal::entityTypeManager()->getViewBuilder('commerce_product_attribute_value') to its own line, to shorten this.